### DISCUSSION TITLE
Move Winograd transforms to upstream LLVM?

### DISCUSSION THREAD

**Discussion by GeorgeARM** [iree-org/iree#16571] [General]
Hello all,

We are interested in moving the Winograd transforms to upstream LLVM so that they can be more broadly reusable.
This though comes with a pack of questions :)

First of all would you be happy with us doing this?
If yes, are there any changes that we need to consider functionally before upstreaming?

Thank you in advance,
George

CC: @nicolasvasilache @MaheshRavishankar @Abhishek-Varma @jpienaar

**Comment by MaheshRavishankar** [comment]
@stellaraccident  can chip in about licenses. 

With respect to moving this. We probably want to take stock of what the op does today. This op is similar to softmax. It should be implementing the `AggregatedOpInterface` as well. It might be better to do that in IREE first and make sure things work before moving it upstream.

**Reply by RoboTux** [reply]
Hi @MaheshRavishankar, could the AggregatedOpInterface added (first in IREE then upstream as suggested) as a second step? It would make for a smaller patch to review and allows to get the winnograd support to upstream quicker.

**Reply by MaheshRavishankar** [reply]
I think thats fine

**Comment by MaheshRavishankar** [comment]
(but thanks a lot for looking into moving it. Would be really good to move it and 100% supportive. Just want to make sure we flush things out before moving it)

**Comment by stellaraccident** [comment]
Nothing special needs to be done to upstream. Just use the upstream license headers. They are compatible.
